projects/ngx-indexed-db/src/lib/ngx-indexed-db.ts

```diff
@@ -196,6 +196,21 @@ export class NgxIndexedDB {
 			};
 		});
 	}
+
+	getAllByIndex(storeName: string, indexName: string, key: any) {
+		return new Promise<any>((resolve, reject) => {
+			this.dbWrapper.validateBeforeTransaction(storeName, reject);
+			let transaction = this.dbWrapper.createTransaction(
+					this.dbWrapper.optionsGenerator(DBMode.readonly, storeName, reject, resolve)
+				),
+				objectStore = transaction.objectStore(storeName),
+				index = objectStore.index(indexName),
+				request = index.getAll(key);
+			request.onsuccess = event => {
```

Comment:
Can you handle the rejection also, please?

Reply:
done :+1: